blocks_vertical/data.js

```diff
@@ -587,12 +639,13 @@ Blockly.Constants.Data.RENAME_OPTION_CALLBACK_FACTORY = function(block) {
  * Callback for delete variable dropdown menu option associated with a
  * variable getter block.
  * @param {!Blockly.Block} block The block with the variable to delete.
+ * @param {string} type The type of the field (VARIABLE or LIST).
  * @return {!function()} A function that deletes the variable.
  */
-Blockly.Constants.Data.DELETE_OPTION_CALLBACK_FACTORY = function(block) {
+Blockly.Constants.Data.DELETE_OPTION_CALLBACK_FACTORY = function(block, type) {
   return function() {
     var workspace = block.workspace;
-    var variable = block.getField('VARIABLE').getVariable();
+    var variable = block.getField(type).getVariable();
     workspace.deleteVariableById(variable.getId());
```

Comment:
It should be noted that this shows a confirmation bubble which displays "Delete (N) uses of this variable?". Changing that to show "list" would require changing `Blockly.VariableMap.prototype.deleteVariableById` (`core/variable_map.js`) - you'd need to check the type of the variable, then use the appropriate `Blockly.Msg` under `var confirmText`.